ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1023,10 +1023,10 @@ extension SignalType {
 
 private struct ZipState<Left, Right> {
 	var values: (left: [Left], right: [Right]) = ([], [])
-	var completed: (left: Bool, right: Bool) = (false, false)
+	var isCompleted: (left: Bool, right: Bool) = (false, false)
 
 	var isFinished: Bool {
-		return (completed.left && values.left.isEmpty) || (completed.right && values.right.isEmpty)
+		return (isCompleted.left && values.left.isEmpty) || (isCompleted.right && values.right.isEmpty) || (isCompleted.left && isCompleted.right && (values.left.isEmpty || values.right.isEmpty))
```

Comment:
The added case here is redundant.

Before this was `(C1 && V1) || (C2 && V2)`. You've made it `(C1 && V1) || (C2 && V2) || (C1 && C2 && (V1 || V2)). If `C1 && C2` is true, then this can be simplified to `V1 || V2 || (V1 || V2)`.